test/test_shared.cpp

```diff
@@ -2307,6 +2307,28 @@ TEST(Shared_MixedWithNonShared)
 #endif
 }
 
+
+#if REALM_ENABLE_ENCRYPTION
+// verify that even though different threads share the same encrypted pages,
+// a thread will not get access without the key.
+TEST(Shared_EncryptionKeyCheck)
+{
+    if (crypt_key() != nullptr) { // only test if encryption is actually in use
+        SHARED_GROUP_TEST_PATH(path);
+        SharedGroup sg(path, false, SharedGroupOptions(crypt_key()));
```

Comment:
You can also use `crypt_key(true)` here and below to always use a key, even if the other unit tests don't.